201-front-door-premium-vm-private-link/modules/load-balancer.bicep

```diff
@@ -0,0 +1,71 @@
+@description('The location into which the load balancer resources should be deployed.')
+param location string
+
+@description('The resource ID of the virtual network subnet that the load balancer should be deployed into.')
+param subnetResourceId string
+
+var loadBalancerName = 'MyLoadBalancer'
+var frontendIPConfigurationName = 'MyFrontendIPConfiguration'
+var healthProbeName = 'MyHealthProbe'
+var backendAddressPoolName = 'MyBackendAddressPool'
+
+resource loadBalancer 'Microsoft.Network/loadBalancers@2020-06-01' = {
+  name: loadBalancerName
+  location: location
+  sku: {
+    name: 'Standard'
+  }
+  properties: {
+    frontendIPConfigurations: [
+      {
+        name: frontendIPConfigurationName
+        properties: {
+          privateIPAllocationMethod: 'Dynamic'
+          subnet: {
+            id: subnetResourceId
+          }
+        }
+      }
+    ]
+    backendAddressPools: [
+      {
+        name: backendAddressPoolName
+      }
+    ]
+    probes: [
+      {
+        name: healthProbeName
+        properties: {
+          protocol: 'Tcp'
+          port: 80
+          intervalInSeconds: 15
+          numberOfProbes: 2
+        }
+      }
+    ]
+    loadBalancingRules: [
+      {
+        name: 'HttpRule'
+        properties: {
+          protocol: 'Tcp'
+          frontendPort: 80
+          backendPort: 80
+          idleTimeoutInMinutes: 15
+          frontendIPConfiguration: {
+            id: resourceId('Microsoft.Network/loadBalancers/frontendIpConfigurations', loadBalancerName, frontendIPConfigurationName)
```

Comment:
have you tried the bicep version of the resourceId here (and below)?

loadBalancer.properties.frontendIpConfigurations.id

Reply:
Yes - that causes a circular reference since this is part of the same resource declaration.

Reply:
do you mind opening an issue in bicep for this?  I get it, but seems solvable (and if we stick with it this way can never get rid of the resourceId fn)

Reply:
Sure: https://github.com/Azure/bicep/issues/1852

I'll keep the current behaviour for this PR though, and can update it if/when Bicep gets this improvement.